### PR TITLE
[NGraph] VisualizeTree denormal visualization

### PR DESCRIPTION
**Pull request by eshoguli**
nGraph `VisualizeTree` `Constant` operation visualization update:
1. values are not visualized - fixed,
2. min/max/denormal values visualization

### CODE REVIEW

File: ngraph/core/src/pass/visualize_tree.cpp [hunk 1]
```diff
@@ -328,27 +329,83 @@ static std::string pretty_partial_shape(const PartialShape& shape)
...
+    const std::string additional_ss = getenv_bool("NGRAPH_VISUALIZE_TREE_MIN_MAX_DENORMAL")
+                                          ? pretty_min_max_denormal_value(values)
+                                          : "";
+
     std::stringstream ss;
     bool first = true;
     for (size_t i = 0; i < values.size(); ++i)
     {
-        if (i > 32)
+        if (i > 31)
```
Comment (ilyachur) [inline ngraph/core/src/pass/visualize_tree.cpp:388]: Why did you change `32` to `31`?

File: ngraph/core/src/pass/visualize_tree.cpp [hunk 2]
```diff
@@ -328,27 +329,83 @@ static std::string pretty_partial_shape(const PartialShape& shape)
...
+        {
+            max_value = value;
+        }
+
+        const auto abs_value = std::abs(static_cast<double>(value));
+        if (((abs_value > 0.) && (abs_value < 1.e-32)) || (abs_value > 1.e+32))
+        {
+            if (denormal_counts < 3)
+            {
+                denormal_ss << (denormal_counts > 0 ? ", " : "") << i << ": " << value;
```
Comment (ilyachur) [inline ngraph/core/src/pass/visualize_tree.cpp:358]: Why do you add only first 3 denormal elements?
What if vector contains not only denormals?

File: ngraph/core/src/pass/visualize_tree.cpp (hunk 1 above)
Comment (eshoguli) [reply ngraph/core/src/pass/visualize_tree.cpp:388]: to have 32 values

File: ngraph/core/src/pass/visualize_tree.cpp (hunk 2 above)
Comment (eshoguli) [reply ngraph/core/src/pass/visualize_tree.cpp:358]: > Why do you add only first 3 denormal elements?  

I don't have any requests from other team members, first 3 elements are enough for me : )
> What if vector contains not only denormals?  

Original values are not affected. Denormal values will be displayed at the end of the original box.

**Review by ilyachur — APPROVED** [review]

File: ngraph/core/src/pass/visualize_tree.cpp [hunk 3]
```diff
@@ -349,19 +350,74 @@ static std::string pretty_partial_shape(const PartialShape& shape)
...
+            {
+                ss << std::endl;
+            }
+        }
+        else
         {
             ss << "...";
+            if (!additional_ss.empty())
+            {
+                ss << std::endl << "(" << additional_ss << ")";
```
Comment (GlebKazantaev) [inline ngraph/core/src/pass/visualize_tree.cpp:419]: It looks very strange. Please move `if (!additional_ss.empty())` after this cycle.
Also I expect that the number of denormal values will be equal to the number of values printed in prety_value function.

**Review by GlebKazantaev — CHANGES_REQUESTED** [review]

File: ngraph/core/src/pass/visualize_tree.cpp (hunk 3 above)
Comment (eshoguli) [reply ngraph/core/src/pass/visualize_tree.cpp]: > It looks very strange. Please move if (!additional_ss.empty()) after this cycle.

Done: fixed, thanks!
> Also I expect that the number of denormal values will be equal to the number of values printed in prety_value function.

We display amount of denormal values and 3 values only. It was enough for me for a model enabling.

Comment (GlebKazantaev) [reply ngraph/core/src/pass/visualize_tree.cpp]: From my point of view we don't need to have another code that traverse vector and convert denormal values. I suggest to have a simple function that works with single value but not with vector.
Also your point that amount of values 3 is enough is a bit confusing. I suggest to limit the number of denormal values in the same way as for other values.

Comment (eshoguli) [reply ngraph/core/src/pass/visualize_tree.cpp]: >From my point of view we don't need to have another code that traverse vector and convert denormal values. I suggest to have a simple function that works with single value but not with vector.

First, we are speaking about visualization, correct? We need to visualize if denormal values exist or not, there is no conversion here. Second, denormal visualization is optional and doesn't affect anything by default.

>Also your point that amount of values 3 is enough is a bit confusing. I suggest to limit the number of denormal values in the same way as for other values.

`Constant` operation is displayed as a box. 3 values and their indexes are enough to fill available space after min/max and the number of denormal values. We need to display some denormal values and their indexes to track these values operation by operation in the dump files.

**Review by GlebKazantaev — APPROVED** [review]
